src/_pytest/hookspec.py

```diff
@@ -261,27 +262,39 @@ def pytest_collection_finish(session: "Session") -> None:
 
 
 @hookspec(firstresult=True)
-def pytest_ignore_collect(path: py.path.local, config: "Config") -> Optional[bool]:
+def pytest_ignore_collect(
+    fspath: Path, path: py.path.local, config: "Config"
+) -> Optional[bool]:
     """Return True to prevent considering this path for collection.
 
     This hook is consulted for all files and directories prior to calling
     more specific hooks.
 
     Stops at first non-None result, see :ref:`firstresult`.
 
+    :param pathlib.Path fspath: The path to analyze.
     :param py.path.local path: The path to analyze.
     :param _pytest.config.Config config: The pytest config object.
+
+    versionchanged:: 6.3.0
+        The ``fspath`` parameter was added as a pathlib.Path equivalent
+        of the ``path`` parameter.
     """
 
 
 def pytest_collect_file(
-    path: py.path.local, parent: "Collector"
+    fspath: Path, path: py.path.local, parent: "Collector"
 ) -> "Optional[Collector]":
     """Create a Collector for the given path, or None if not relevant.
 
     The new node needs to have the specified ``parent`` as a parent.
 
+    :param pathlib.Path fspath: The path to analyze.
     :param py.path.local path: The path to collect.
+
+    versionchanged:: 6.3.0
+        The ``fspath`` parameter was added as a pathlib.Path equivalent
```

Comment:
I _think_ this works:

```suggestion
        The ``fspath`` parameter was added as a :class:`pathlib.Path` equivalent
```